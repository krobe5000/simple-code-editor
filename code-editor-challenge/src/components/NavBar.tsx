import React from "react";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import { LoginLogoutButton } from "./LoginLogoutButton";
import Profile from "../components/Profile";
import Grid from "@material-ui/core/Grid";
import OpenWorkspaceButton from "./OpenWorkspaceButton";

const NavBar = () => {
  return (
    <AppBar position="static">
      <Toolbar>
        <Grid container spacing={3}>
          <Grid item xs={12} />
          <Grid item xs={4} />
          <Grid item xs={2}>
            <OpenWorkspaceButton />
          </Grid>
          <Grid item xs={2}>
            <LoginLogoutButton />
          </Grid>
          <Grid item xs={2}>
            <Profile />
          </Grid>
        </Grid>
      </Toolbar>
    </AppBar>
  );
};

export default NavBar;
