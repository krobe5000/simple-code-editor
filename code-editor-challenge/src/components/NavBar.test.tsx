import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import NavBar from "./NavBar";

const renderComponent = () => {
  return render(<NavBar />);
};

test("Should render some buttons", () => {
  const container = renderComponent();

  expect(container).toMatchSnapshot();
});
