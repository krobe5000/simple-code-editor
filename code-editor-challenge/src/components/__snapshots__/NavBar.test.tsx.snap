// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render some buttons 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            />
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
            />
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
            >
              <label
                for="filePicker"
                style="color: rgb(255, 255, 255); background-color: rgb(245, 0, 87); padding: 10px 16px; box-shadow: 0px 3px 1px -2px rgb(0 0 0 / 20%), 0px 2px 2px 0px rgb(0 0 0 / 14%), 0px 1px 5px 0px rgb(0 0 0 / 12%); border-radius: 4px; font-family: Roboto, Helvetica, Arial, sans-serif; font-size: 0.875rem; font-weight: bold; min-width: 64px; box-sizing: border-box; height: 36px; cursor: pointer; display: inline-flex;"
              >
                Open Workspace
              </label>
              <input
                accept=".tsx .ts .pdf .json"
                directory=""
                id="filePicker"
                style="opacity: 0;"
                type="file"
                webkitdirectory=""
              />
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Log In
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
            />
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          />
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
          />
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
          >
            <label
              for="filePicker"
              style="color: rgb(255, 255, 255); background-color: rgb(245, 0, 87); padding: 10px 16px; box-shadow: 0px 3px 1px -2px rgb(0 0 0 / 20%), 0px 2px 2px 0px rgb(0 0 0 / 14%), 0px 1px 5px 0px rgb(0 0 0 / 12%); border-radius: 4px; font-family: Roboto, Helvetica, Arial, sans-serif; font-size: 0.875rem; font-weight: bold; min-width: 64px; box-sizing: border-box; height: 36px; cursor: pointer; display: inline-flex;"
            >
              Open Workspace
            </label>
            <input
              accept=".tsx .ts .pdf .json"
              directory=""
              id="filePicker"
              style="opacity: 0;"
              type="file"
              webkitdirectory=""
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Log In
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
          />
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
